File: ObjectivelyHybridiOSExample/ViewController.m
/*
 Copyright (c) 2014 Lee Barney
 Permission is hereby granted, free of charge, to any person obtaining a
 copy of this software and associated documentation files (the "Software"),
 to deal in the Software without restriction, including without limitation the
 rights to use, copy, modify, merge, publish, distribute, sublicense,
 and/or sell copies of the Software, and to permit persons to whom the Software
 is furnished to do so, subject to the following conditions:
 
 The above copyright notice and this permission notice shall be
 included in all copies or substantial portions of the Software.
 
 
 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
 CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
 OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 
 */

#import "ViewController.h"

@interface ViewController ()
@property (strong, nonatomic)WKWebView *appWebView;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    WKWebViewConfiguration *theConfiguration = [[WKWebViewConfiguration alloc] init];
    [theConfiguration.userContentController addScriptMessageHandler:self name:@"interOp"];
    _appWebView = [[WKWebView alloc] initWithFrame:self.view.frame configuration:theConfiguration];

    NSString *path = [[NSBundle mainBundle] pathForResource:@"index" ofType:@"html"];
    NSURL *url = [NSURL fileURLWithPath:path];
    NSURLRequest *request = [NSURLRequest requestWithURL:url];
    [_appWebView loadRequest:request];
    [self.view addSubview:_appWebView];

}
/*
 * Change the behavior of this method to match what you need for your app.
 */
-(void)userContentController:(WKUserContentController *)userContentController didReceiveScriptMessage:(WKScriptMessage *)message{
    NSDictionary * sentData = (NSDictionary*)message.body;
    int aCount = [sentData[@"count"] intValue];
    [_appWebView evaluateJavaScript:[NSString stringWithFormat:@"storeAndShow(%d)",aCount + 1] completionHandler:^(id JSReturnValue, NSError* error){
        if (error) {
            NSLog(@"error: %@", error.description);
        }
        else if (JSReturnValue != nil){
            NSLog(@"returned value: %@",JSReturnValue);
        }
        else{
            NSLog(@"no return from JS");
        }
    }];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: ObjectivelyHybridiOSExample/ViewController.swift
 import UIKit
import WebKit

class ViewController: UIViewController {
    lazy var webView: WKWebView = {
        let configuration = WKWebViewConfiguration()
        configuration.userContentController.add(self, name: "SOFA")

        let view = WKWebView(frame: self.view.frame, configuration: configuration)
        view.scrollView.isScrollEnabled = true
        view.autoresizingMask = [.flexibleWidth, .flexibleHeight]

        return view
    }()

    override func viewDidLoad() {
        super.viewDidLoad()

        self.view.addSubview(self.webView)

        let path = Bundle.main.path(forResource: "index", ofType: "html")!
        let url = URL(fileURLWithPath: path)
        let request = URLRequest(url: url)
        self.webView.load(request)
    }
}

extension ViewController: WKScriptMessageHandler {
    func userContentController(_ userContentController: WKUserContentController, didReceive message: WKScriptMessage) {
        if let sentData = message.body as? [String: Any] {
            if let count = sentData["count"] as? Int {
                self.webView.evaluateJavaScript("storeAndShow(\(count + 1))") { jsReturnValue, error in
                    if let error = error {
                        print("Error: \(error)")
                    } else if let newCount = jsReturnValue as? Int {
                        print("Returned value: \(newCount)")
                    } else {
                        print("No return from JS")
                    }
                }
            }
        }
    }
}
